xbmc/interfaces/json-rpc/AddonsOperations.cpp

```diff
@@ -79,19 +80,29 @@ JSONRPC_STATUS CAddonsOperations::GetAddons(const std::string &method, ITranspor
     if (*typeIt == ADDON_UNKNOWN)
     {
       if (!enabled.isBoolean()) //All
-        CAddonMgr::GetInstance().GetInstalledAddons(typeAddons);
-      else if (enabled.asBoolean()) //Enabled
+      {
+        if (!installed.isBoolean() || installed.asBoolean())
+          CAddonMgr::GetInstance().GetInstalledAddons(typeAddons);
+        if (!installed.isBoolean() || (installed.isBoolean() && !installed.asBoolean()))
+          CAddonMgr::GetInstance().GetInstallableAddons(typeAddons);
+      }
+      else if (enabled.asBoolean() && (!installed.isBoolean() || installed.asBoolean())) //Enabled
         CAddonMgr::GetInstance().GetAddons(typeAddons);
-      else
+      else if (!installed.isBoolean() || installed.asBoolean())
```

Comment:
I'm not sure if this makes sense. There are no not-installed disabled add-ons so the question is whether the "installed" parameter makes any sense when "enabled" is false. But this way it is consistent.